sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/security/HardcodedCredentialsCheck.java

```diff
@@ -118,7 +118,15 @@ private void checkAttributes(Node node, Set<String> credentialWords, boolean rep
 
   private static boolean isCredentialNode(Node node, Set<String> credentialWords) {
     String localName = node.getLocalName();
-    return localName != null && credentialWords.contains(localName.toLowerCase(Locale.ROOT));
+    if (localName == null || !credentialWords.contains(localName.toLowerCase(Locale.ROOT))) {
+      return false;
+    }
+    return !isTruthyAndroidPasswordAttribute(node);
+  }
+
+  private static boolean isTruthyAndroidPasswordAttribute(Node node) {
+    return node.getNodeName().equalsIgnoreCase("android:password") &&
+      node.getNodeValue().trim().equalsIgnoreCase("true");
```

Comment:
The rule aims to detect hardcoded credentials, this is not the case for `android:password` regardless of the value, we can probably remove this condition.

Reply:
Indeed, the ticket has been modified to reflect that